pointcloud/basic_octree.go

```diff
@@ -182,3 +214,63 @@ func (octree *BasicOctree) CollidesWithGeometry(geom spatialmath.Geometry, thres
 	}
 	return false, errors.New("unknown octree node type")
 }
+
+// CollidesWith checks if the given octree collides with the given geometry and returns true if it does.
+func (octree *BasicOctree) CollidesWith(geom spatialmath.Geometry) (bool, error) {
+	return octree.CollidesWithGeometry(geom, confidenceThreshold, buffer)
+}
+
+// DistanceFrom returns the distance from the given octree to the given geometry.
+func (octree *BasicOctree) DistanceFrom(geom spatialmath.Geometry) (float64, error) {
+	// TODO (RSDK-3743): currently implemented as the bare minimum but needs to be changed to correct implementation
+	collides, err := octree.CollidesWith(geom)
+	if err != nil {
+		return math.Inf(1), err
+	}
+	if collides {
+		return -1, nil
+	}
+	return 1, nil
+}
+
+// EncompassedBy returns true if the given octree is within the given geometry.
+func (octree *BasicOctree) EncompassedBy(geom spatialmath.Geometry) (bool, error) {
+	// TODO (RSDK-3743)
+	return false, errors.New("not implemented")
+}
+
+// SetLabel sets the label of this octree.
+func (octree *BasicOctree) SetLabel(label string) {
+	octree.label = label
+}
+
+// Label returns the label of this octree.
+func (octree *BasicOctree) Label() string {
+	return octree.label
+}
+
+// String returns a human readable string that represents this octree.
+func (octree *BasicOctree) String() string {
```

Comment:
[opt] can you add a note saying that any of the octree's children will not be represented in this string.